eslint/babel-eslint-plugin/test/rules/no-invalid-this.js

```diff
@@ -78,40 +77,76 @@ function extractPatterns(patterns, type) {
 //------------------------------------------------------------------------------
 
 const patterns = [
-  // Class private fields
   {
-    code: "class A {a = this.b;};",
-    parserOptions: { ecmaVersion: 6 },
-    valid: [NORMAL, USE_STRICT, IMPLIED_STRICT, MODULES],
+    code: "class A { a = this.b; }",
+    parserOptions: { ecmaVersion: 2020 },
+    valid: [MODULE, SCRIPT, USE_STRICT, IMPLIED_STRICT],
     invalid: [],
   },
-
   {
-    code: "class A {a = () => {return this.b;};};",
-    parserOptions: { ecmaVersion: 6 },
-    valid: [NORMAL, USE_STRICT, IMPLIED_STRICT, MODULES],
+    code: "class A { a = () => { return this.b; }; }",
+    parserOptions: { ecmaVersion: 2020 },
+    valid: [MODULE, SCRIPT, USE_STRICT, IMPLIED_STRICT],
     invalid: [],
   },
-
-  // Class Private methods
   {
-    code: "class A {#a = this.b;};",
-    parserOptions: { ecmaVersion: 6 },
-    valid: [NORMAL, USE_STRICT, IMPLIED_STRICT, MODULES],
+    code: "class A { a = () => this.b; }",
+    parserOptions: { ecmaVersion: 2020 },
+    valid: [MODULE, SCRIPT, USE_STRICT, IMPLIED_STRICT],
     invalid: [],
   },
-
   {
-    code: "class A {#a = () => {return this.b;};};",
-    parserOptions: { ecmaVersion: 6 },
-    valid: [NORMAL, USE_STRICT, IMPLIED_STRICT, MODULES],
+    code: "class A { [this] = b; }",
+    parserOptions: { ecmaVersion: 2020 },
+    valid: [],
+    invalid: [MODULE, SCRIPT, USE_STRICT, IMPLIED_STRICT],
+  },
+  {
+    code: "class A { [this.a] = b; }",
+    parserOptions: { ecmaVersion: 2020 },
+    valid: [],
+    invalid: [MODULE, SCRIPT, USE_STRICT, IMPLIED_STRICT],
+  },
+  {
+    code: "class A { [`${this.a}b`] = c; }",
+    parserOptions: { ecmaVersion: 2020 },
+    valid: [],
+    invalid: [MODULE, SCRIPT, USE_STRICT, IMPLIED_STRICT],
+  },
+  {
+    code: "class A { [this.a] = () => { return b; }; }",
+    parserOptions: { ecmaVersion: 2020 },
+    valid: [],
+    invalid: [MODULE, SCRIPT, USE_STRICT, IMPLIED_STRICT],
+  },
+  {
+    code: "class A { [this.a] = () => b; }",
```

Comment:
Can you add a case of nested class? The class A can be nested in i.e. constructor of class B so this becomes valid.

Sorry I can't draft an example on mobile. I hope you can get it. 😆

Reply:
Good call!

Reply:
Does this cover everything?